src/Query/DateParser.ts

```diff
@@ -75,12 +90,42 @@ export class DateParser {
             const unitOfTime = range === 'week' ? 'isoWeek' : (range as moment.unitOfTime.DurationConstructor);
             dateRange = [dateRange[0].startOf(unitOfTime), dateRange[1].endOf(unitOfTime)];
 
-            // Dates shall be at midnight eg 00:00
-            dateRange.forEach((d) => d.startOf('day'));
-
-            return dateRange;
+            return DateParser.setDateRangeToStartOfDay(dateRange);
         }
 
         return undefined;
     }
+
+    private static parseSpecificDateRange(input: string): [moment.Moment, moment.Moment] | undefined {
+        let parsedRange: [moment.Moment, moment.Moment] | undefined = undefined;
+
+        const parsingVectors: [RegExp, string, moment.unitOfTime.StartOf][] = [
+            [/[0-9]{4}/, 'YYYY', 'year'],
+            [/[0-9]{4}-Q[1-4]/, 'YYYY-Q', 'quarter'],
+            [/[0-9]{4}-[0-9]{2}/, 'YYYY-MM', 'month'],
+            [/[0-9]{4}-W[0-9]{2}/, 'YYYY-WW', 'isoWeek'],
+        ];
+
+        parsingVectors.every((vector) => {
+            const regexp = vector[0];
+            const format = vector[1];
+            const unit = vector[2];
```

Comment:
This is all looking really lovely now.

An "either you know it or you don't" thing is to replace the `every()` or `forEach()` with a for loop that returns the first found value...


```diff
-        parsingVectors.every((vector) => {
-            const regexp = vector[0];
-            const format = vector[1];
-            const unit = vector[2];
+        for (const [regexp, format, unit] of parsingVectors) {
```

You will find that you can reduce the scope of the `parsedRange` to inside the for loop, and return as soon as you have found a matching value...

(Reducing the scope of variables aids readability a lot - you don't need to hold so much in your head whilst reading the code)

And then at the end you can return `undefined`... 

It simplifies the code even more... Rather more self-explanatory - No need to reason about what confusing `true` and `false` values are doing.

Reply:
> It simplifies the code even more... Rather more self-explanatory - No need to reason about what confusing `true` and `false` values are doing.


I think you probably beat me to implementation that!